File: backend/src/application/config_manager.py
from typing import Dict, Any
from src.infrastructure.workers.worker_config import get_worker_config, WorkerConfig
from src.infrastructure.workers.factories.worker_factory import WorkerFactory

class ConfigManager:
    def __init__(self, initial_config: Dict[str, Any]):
        self.config = initial_config
        self.worker_config = get_worker_config(self.config['worker'])
        self.worker_factory = WorkerFactory(self.worker_config)

    def update_config(self, new_config: Dict[str, Any]):
        self.config.update(new_config)
        self.worker_config = get_worker_config(self.config['worker'])
        self.worker_factory = WorkerFactory(self.worker_config)

    def get_worker_factory(self) -> WorkerFactory:
        return self.worker_factory

    def get_current_config(self) -> Dict[str, Any]:
        return self.config

File: golang/devops_console/internal/application/config_manager.go
package application

import (
	"devops_console/internal/infrastructure/workers"
	"devops_console/internal/infrastructure/workers/factories"
)

type ConfigManager struct {
	config        map[string]interface{}
	workerConfig  workers.WorkerConfig
	workerFactory factories.WorkerFactory
}

func NewConfigManager(initialConfig map[string]interface{}) *ConfigManager {
	workerConfig := workers.GetWorkerConfig(initialConfig["worker"])
	return &ConfigManager{
		config:        initialConfig,
		workerConfig:  workerConfig,
		workerFactory: factories.NewWorkerFactory(workerConfig),
	}
}

func (cm *ConfigManager) UpdateConfig(newConfig map[string]interface{}) {
	for k, v := range newConfig {
		cm.config[k] = v
	}
	cm.workerConfig = workers.GetWorkerConfig(cm.config["worker"])
	cm.workerFactory = factories.NewWorkerFactory(cm.workerConfig)
}

func (cm *ConfigManager) GetWorkerFactory() factories.WorkerFactory {
	return cm.workerFactory
}

func (cm *ConfigManager) GetCurrentConfig() map[string]interface{} {
	return cm.config
}
